### PR TITLE
enh(ui): Add diverses  formats option to export the graphs

### PR DESCRIPTION
**Pull request by Jordan91**
## Description

Allow for users to use pre-determined export sizes:

- Original size
- Medium size
- Small size

Ideally, we'll decide on the standard sizes based on the most common resolutions (1080p, 1440p, 2160p) and feedback from Experts/Clients/Community on their preferred resolution (==> Medium)

## Type of change

- [x] New functionality (non-breaking change)

## Target serie

- [x] 21.04.x
- [x] 21.10.x (master)

<h2> How this pull request can be tested ? </h2>

- On resource status, click on service with graph available, 
then on detail panel on tab GRAPH , 
- Click on icon to export the graph and select size you want export for graph.

Demo:
https://www.loom.com/share/921a623a600c4398a3b40444365358d7

### CODE REVIEW

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -16139,4 +16139,13 @@ msgid "At least one column must be selected"
...
+msgstr "La durée doit être inférieure à une année"
+
+#  msgid "Current"
+#  msgstr "Taille courante"
+
+#  msgid "Medium"
+#  msgstr "Taille moyenne"
+
+#  msgid "Small"
+#  msgstr "Taille petite"
```
Comment (Thebarda) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16151]:
```suggestion
msgid "Current"
msgstr "Courante"

msgid "Medium"
msgstr "Moyenne"

msgid "Small"
msgstr "Petite"
```

Comment (Thebarda) [reply lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16151]: Or change `Current` to `Current size`, etc...

File: www/front_src/src/Resources/translatedLabels.ts [hunk 2]
```diff
@@ -213,3 +213,7 @@ export const labelCalculationType = 'Calculation type';
 export const labelSelectAtLeastOneColumn =
   'At least one column must be selected';
 export const labelMaxDuration1Year = 'The duration must be lesser than a year';
+export const labelSelectedSizeExport = 'Selected Size Export';
```
Comment (Thebarda) [inline www/front_src/src/Resources/translatedLabels.ts:216]: This label is not used here. Is it for test purpose ?

Comment (Jordan91) [reply www/front_src/src/Resources/translatedLabels.ts:216]: No it's on my own, Bruno disagree this change.

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po (hunk 1 above)
Comment (Jordan91) [reply lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16151]: Hi, ok i'll push this change

**Comment by bdauria** [comment]
Original size makes more sense than Current size in my opinion

**Comment by karimsx** [comment]
hey @Jordan91 i found this traductor that using AI if it can help for other language such as es/de ...

https://www.deepl.com/translator

**Comment by Jordan91** [comment]
> Original size makes more sense than Current size in my opinion

Yes i understand , it's done

**Comment by Jordan91** [comment]
> hey @Jordan91 i found this traductor that using AI if it can help for other language such as es/de ...
> 
> https://www.deepl.com/translator

Thank's ! If i understood well , this translation is in charge by another colleague (spanish men) but thank's for the links ;)

File: www/front_src/src/Resources/translatedLabels.ts [hunk 3]
```diff
@@ -213,3 +213,6 @@ export const labelCalculationType = 'Calculation type';
 export const labelSelectAtLeastOneColumn =
   'At least one column must be selected';
 export const labelMaxDuration1Year = 'The duration must be lesser than a year';
+export const labelOriginalSizeExport = 'Original';
+export const labelMediumSizeExport = 'Medium';
+export const labelSmallSizeExport = 'Small';
```
Comment (bdauria) [inline www/front_src/src/Resources/translatedLabels.ts:218]:
```suggestion
export const labelOriginal = 'Original';
export const labelMedium = 'Medium';
export const labelSmall = 'Small';
```

File: www/front_src/src/Resources/Graph/Performance/index.tsx [hunk 4]
```diff
@@ -378,14 +397,32 @@ const PerformanceGraph = ({
...
+                    onClose={closeSizeExportMenu}
+                  >
+                    <MenuItem onClick={() => convertToPng(1)}>
+                      {t(labelOriginalSizeExport)}
+                    </MenuItem>
+                    <MenuItem onClick={() => convertToPng(0.75)}>
+                      {t(labelMediumSizeExport)}
+                    </MenuItem>
+                    <MenuItem onClick={() => convertToPng(0.5)}>
+                      {t(labelSmallSizeExport)}
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/index.tsx:422]:
```suggestion
                      {t(labelOriginal)}
                    </MenuItem>
                    <MenuItem onClick={() => convertToPng(0.75)}>
                      {t(labelMedium)}
                    </MenuItem>
                    <MenuItem onClick={() => convertToPng(0.5)}>
                      {t(labelSmall)}
```

File: www/front_src/src/Resources/Graph/Performance/index.tsx [hunk 5]
```diff
@@ -40,7 +46,10 @@ import { Resource } from '../../models';
 import { ResourceDetails } from '../../Details/models';
 import { CommentParameters } from '../../Actions/api';
 import {
+  labelOriginalSizeExport,
   labelExportToPng,
+  labelSmallSizeExport,
+  labelMediumSizeExport,
```
Comment (bdauria) [inline www/front_src/src/Resources/Graph/Performance/index.tsx:52]:
```suggestion
  labelOriginal,
  labelExportToPng,
  labelSmall,
  labelMedium,
```

File: www/front_src/src/Resources/translatedLabels.ts (hunk 3 above)
Comment (Jordan91) [reply www/front_src/src/Resources/translatedLabels.ts]: done !

File: www/front_src/src/Resources/Graph/Performance/index.tsx (hunk 4 above)
Comment (Jordan91) [reply www/front_src/src/Resources/Graph/Performance/index.tsx]: done !

File: www/front_src/src/Resources/Graph/Performance/index.tsx (hunk 5 above)
Comment (Jordan91) [reply www/front_src/src/Resources/Graph/Performance/index.tsx]: Done !

**Review by Thebarda — DISMISSED** [review]

**Comment by bdauria** [comment]
@Jordan91 i think it might make more sense to add "size" after the labels to be more clear on the action. For instance : Original size (Taille originale)

We should also remove the deprecated translations (Current)

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -14706,4 +14706,14 @@ msgstr "Permitir certificado autofirmado"
 #  msgstr ""
 
 #  msgid "The duration must be lesser than a year"
+#  msgstr ""
+
+#  msgid "As Displayed"
```
Comment (bdauria) [inline lang/es_ES.UTF-8/LC_MESSAGES/messages.po:14711]:
```suggestion
#  msgid "As displayed"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -16139,4 +16139,13 @@ msgid "At least one column must be selected"
 msgstr "Au moins une colonne doit être sélectionnée"
 
 msgid "The duration must be lesser than a year"
-msgstr "La durée doit être inférieure à une année"
\ No newline at end of file
+msgstr "La durée doit être inférieure à une année"
+
+#  msgid "As Displayed"
```
Comment (bdauria) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16144]:
```suggestion
#  msgid "As displayed"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 8]
```diff
@@ -16139,4 +16139,13 @@ msgid "At least one column must be selected"
...
+#  msgstr "Comme Affiché"
```
Comment (bdauria) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16145]:
```suggestion
#  msgstr "Comme affiché"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 9]
```diff
@@ -16139,4 +16139,13 @@ msgid "At least one column must be selected"
...
+
+#  msgid "Medium size"
+#  msgstr "Taille Moyenne"
```
Comment (bdauria) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16148]:
```suggestion
#  msgstr "Taille moyenne"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 10]
```diff
@@ -16139,4 +16139,13 @@ msgid "At least one column must be selected"
...
+
+#  msgid "Small size"
+#  msgstr "Petite Taille"
```
Comment (bdauria) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16151]:
```suggestion
#  msgstr "Petite taille"
```

File: lang/pt_PT.UTF-8/LC_MESSAGES/messages.po [hunk 11]
```diff
@@ -15145,4 +15145,14 @@ msgstr "Erro ao remover tokens de um contato"
 #  msgstr ""
 
 #  msgid "The duration must be lesser than a year"
+#  msgstr ""
+
+#  msgid "As Displayed"
```
Comment (bdauria) [inline lang/pt_PT.UTF-8/LC_MESSAGES/messages.po:15150]:
```suggestion
#  msgid "As displayed"
```

File: www/front_src/src/Resources/translatedLabels.ts [hunk 12]
```diff
@@ -213,3 +213,6 @@ export const labelCalculationType = 'Calculation type';
 export const labelSelectAtLeastOneColumn =
   'At least one column must be selected';
 export const labelMaxDuration1Year = 'The duration must be lesser than a year';
+export const labelAsDisplayed = 'As Displayed';
```
Comment (bdauria) [inline www/front_src/src/Resources/translatedLabels.ts:216]:
```suggestion
export const labelAsDisplayed = 'As displayed';
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 13]
```diff
@@ -16139,4 +16139,13 @@ msgid "At least one column must be selected"
...
+msgstr "La durée doit être inférieure à une année"
+
+#  msgid "As displayed"
+#  msgstr "Comme affiché"
+
+#  msgid "Medium size"
+#  msgstr "Taille moyenne"
+
+#  msgid "Small size"
+#  msgstr "Petite Taille"
```
Comment (Thebarda) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16151]:
```suggestion
msgid "As displayed"
msgstr "Comme affiché"

msgid "Medium size"
msgstr "Taille moyenne"

msgid "Small size"
msgstr "Petite taille"
```

**Review by bdauria — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]

**Pull request merged**